src/ProxyClient/Cloudflare.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient;
+
+use FOS\HttpCache\ProxyClient\Invalidation\ClearCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\PurgeCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\TagCapable;
+
+/**
+ * Cloudflare HTTP cache invalidator.
+ *
+ * Additional constructor options:
+ * - zone_identifier       Identifier for your Cloudflare zone you want to purge the cache for
+ * - authentication_token  API authorization token, requires Zone.Cache Purge permissions
+ *
+ * @author Simon Jones <simon@studio24.net>
+ */
+class Cloudflare extends HttpProxyClient implements ClearCapable, PurgeCapable, TagCapable
+{
+    /**
+     * @see https://api.cloudflare.com/#getting-started-endpoints
+     */
+    private const API_ENDPOINT = 'https://api.cloudflare.com/client/v4';
+
+    /**
+     * Batch URL purge limit.
+     *
+     * @see https://api.cloudflare.com/#zone-purge-files-by-url
+     */
+    private const URL_BATCH_PURGE_LIMIT = 30;
+
+    /**
+     * Array of data to send to Cloudflare for purge by URLs request.
+     *
+     * @var array
+     */
+    private $purgeByUrlsData = [];
+
+    /**
+     * JSON encode data.
+     *
+     * @param $data
+     *
+     * @return string
+     *
+     * @throws InvalidArgumentException
+     */
+    protected function encode($data)
+    {
+        $json = json_encode($data, JSON_UNESCAPED_SLASHES);
+        if (false === $json) {
+            throw new \InvalidArgumentException(sprintf('Cannot encode "$data": %s', json_last_error_msg()));
+        }
+
+        return $json;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
```

Comment:
might want to add "Tag invalidation only available with cloudflare enterprise account" here as well.